scripts/hooks/convert_images_hook.py

```diff
@@ -46,31 +61,51 @@ def parse_args():
         dest="trigger_size",
         help="Specify minimum file size to trigger the hook.",
     )
+    argparser.add_argument(
```

Comment:
I think I miscommunicated on your previous PR, I meant to say let's keep both the files `scripts/local/convert_images.py` version and this hook version, the difference being:

- The hook code accepts only `--filenames` arg which is passed from the pre-commit script (which already filters image files from [here](https://github.com/Udayraj123/OMRChecker/blob/93e749/.pre-commit-config.yaml#L11))
- The `local/convert_images.py` script would be used by the Users manually and thus is more generic with support for pdf, tiff etc. It will support input/output dir options as you've already added. You should also support in-place file modification under some flag.
- The hook may import some util functions from the local script. 

One way is to expose a function that accepts a list of filepaths and returns the converted images serially, another function that writes these images serially and lastly a wrapper function to do the os.walk and populates a list of filepaths to pass to the first function. Let me know if this makes sense

ref: starter code here to modify -
https://github.com/Udayraj123/OMRChecker/blob/dev/scripts/local/convert_images.py

Reply:
So you want to keep two different version of image conversion, a hook and a local script ( my previous code i am assuming) and want me to make changes in both the file so as to reuse any utility function or create a wrapper function 

pls verify and correct me so i can make the requested changes

Reply:
That sounds correct. The hook version will be limited to the existing arguments that it is using. The local script version will be more generic. And we will reuse as much functions as we can - that's the idea

Reply:
ok kinda got it i will work on this tommorow

Reply:
Cool! You got it 👍

Reply:
I have made changes to the hooks and made a seprate file for the functioning of bulk conversion pls take a look